File: external-libs/core/src/main/java/com/serhii/core/utils/CollectionUtils.java
package com.serhii.core.utils;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Iterator;
import java.util.Map;

/**
 *  Library java collection helper functions
 */

public class CollectionUtils {

    public static <T> T getLastElement(Collection<T> collection) {

        if ( collection.isEmpty()) {
            return null;
        }

        final Iterator<T> iter = collection.iterator();
        T elem = iter.next();

        while (iter.hasNext()) {
            elem = iter.next();
        }

        return elem;
    }

    public static <T1, T2> T1 getMapKeyByValue(Map<T1, T2> collection, T2 value) {

        for (Map.Entry<T1, T2> entry : collection.entrySet()) {
            if (entry.getValue().equals(value)) {
                return entry.getKey();
            }
        }

        return null;
    }

    public static String[] getConvertedArray(ArrayList<String> collection) {

        String[] array = new String[collection.size()];

        for (int i = 0; i < array.length; i++) {
            array[i] = collection.get(i);
        }

        return (array.length == 0) ? null : array;
    }
}

File: external-libs/android-core/src/main/java/com/serhii/core/utils/CollectionUtils.kt
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */
package com.serhii.core.utils

/**
 * Collection of helper functions
 */
object CollectionUtils {

    fun <T> getLastElement(collection: Collection<T>): T? {
        if (collection.isEmpty()) {
            return null
        }
        val iter = collection.iterator()
        var elem = iter.next()
        while (iter.hasNext()) {
            elem = iter.next()
        }
        return elem
    }

    fun <T1, T2> getMapKeyByValue(collection: Map<T1, T2>, value: T2): T1? {
        for ((key, value1) in collection) {
            if (value1 == value) {
                return key
            }
        }
        return null
    }

}